### ISSUE TITLE
Using code coverage causes freezing

### ISSUE THREAD

**Issue by squarerootfury** [sebastianbergmann/phpunit#1655]
When I use the code coverage functionality of PHPUnit, the program gets stuck. No tests are executed.

I'm using `PHP 5.5.20-1~dotdeb.1` on Debian 7 Wheezy. phpUnit is installed in Version 5.4.0. I've already reinstalled XDebug and phpUnit, but this was no solution.

When I dont use the code coverage functionality, the tests are runned.

**Comment by nurikabe** [comment]
Same.  PHP 5.6.8, PHPUnit 4.6.6

**Comment by sebastianbergmann** [comment]
@squarerootfury Can you elaborate what you mean by "stuck"? Does the process run indefinitely without printing anything and hogging CPU? Which version of Xdebug are you using?

@nurikabe Are you also using Debian?

Please note that this [should](https://github.com/sebastianbergmann/phpunit/blob/master/CONTRIBUTING.md) have been reported on the [php-code-coverage](https://github.com/sebastianbergmann/php-code-coverage) project instead.

**Comment by squarerootfury** [comment]
@sebastianbergmann Situation is the following: I have an configuration for a multi file testing. When CC is disabled, everything runs fine. The tests getting executed. But when I enable the CC, no test gets executed.

I noticed following: When I enabled CC, no PHP process gets executed.

The Version of Xdebug is 5.5.20-1~dotdeb.1

If you want to I can close this issue and move it to the correct project (I did not know that the cc part is separated in its own project, my fault).

**Comment by sebastianbergmann** [comment]
"When I enabled CC, no PHP process gets executed." does not make sense to me. What does happen exactly when you run your testsuite with code coverage enabled?

**Comment by squarerootfury** [comment]
I see following output:

```
╰─$ phpunit -c ./test.xml                                                                                                       1 ↵

PHPUnit 4.5.0 by Sebastian Bergmann and contributors.

Configuration read from /var/www/rd/test.xml
```
When I look into pstree, I see that there is no php process

This is the configuration file `test.xml`:

```
<phpunit bootstrap="./Includes/Router.inc.php">
<testsuites>
  <testsuite name="Project Redesign test suite">
    <directory>./Tests/</directory>
  <exclude>./Includes/DataBase/DBLayer.class.php</exclude>
  </testsuite>
</testsuites>
 <filter>
        <blacklist>
            <file>./Includes/DataBase/DBLayer.class.php</file>
        </blacklist>
    </filter>
<php>
        <server name="SERVER_NAME" value="localhost"/>
</php>
<logging>
  <log type="coverage-html" target="/var/www/rd/cc/"/>
</logging>
</phpunit>
```
When I remove the <logging>-Node, the tests being runned.
I wont exclude that this problem could also a "layer-8 made" problem..

**Comment by sebastianbergmann** [comment]
Does the problem exist with the current version of PHPUnit? How did you install PHPUnit?

It does not make any sense at all, though, that you do not see a PHP process running.

**Comment by squarerootfury** [comment]
I am using the version right from the website (the phar package).

I think I found the reason.

My version of phpUnit was slightly old (4.5 instead of 4.7.4). I updated it and version 4.7.4 told me that 
`Warning:        No whitelist configured for code coverage`. That message was new to me.

I've added the whitelist to the configuration, and it worked. But something is strange at this. I'm having a filter to blacklist a file. I assumed that every non excluded file gets automatically whitelisted. So this was my fault :disappointed: 

Sorry for the inconvenience I may have caused. I'm closing this issue.

@nurikabe Maybe your tests got stuck because of the same reason.
